server/src/main/java/com/coincoinche/events/PlayerBadeEvent.java

```diff
@@ -0,0 +1,46 @@
+package com.coincoinche.events;
+
+import com.coincoinche.engine.BiddingMove;
+import com.coincoinche.engine.cards.Suit;
+
+public class PlayerBadeEvent extends Event {
+  private BiddingMove.Special special;
+  private int value;
+  private Suit suit;
+
+  /**
+   * Event sent by the client when the player did a special bidding during the bidding phase of the
+   * round.
+   *
+   * @param special - the value of the special bidding.
+   */
+  public PlayerBadeEvent(BiddingMove.Special special) {
```

Comment:
So, if I understand correctly, we never really instantiate this class in the backend? This is done automatically by `spring-websockets` when the client sends a message?

Reply:
Yes, that's done in the controller when you see `@Payload PlayerBadeEvent event`.
When the client calls this endpoint, spring parses directly the json into an instance of PlayerBadeEvent, which is then transformed by the controller into a BiddingMove using `BiddingMove.fromEvent(event)`.